flask/templating.py

```diff
@@ -101,7 +101,20 @@ def list_templates(self):
 
 
 def _render(template, context, app):
-    """Renders the template and fires the signal"""
+    """Renders the template and fires signals.
+
+    It is possible to override render template in the before_render_template signal.
+    It can be done only if exactly one receiver and it return not None result."""
+
+    brt_resp = before_render_template.send(app, template=template, context=context)
+    
+    overrides = [rv for _, rv in brt_resp if rv is not None]
```

Comment:
Hm... I guess it would trigger the signal again, and if it's not written properly it would result in an endless loop.

However, I'd assume any function subscribing to this signal would not override *any* template, so if it's called again for a template it renders to override one, it'd be expected not to intercept this call, too.